### PR TITLE
test: replace embedded Git repository with dynamically created repository

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Currently, Trivy includes a separate Git bare repository committed within its codebase for testing purposes. For example: https://github.com/aquasecurity/trivy/tree/29615be85e8bfeaf5a0cd51829b1898c55fa4274/pkg/plugin/testdata/test_plugin.git

While this approach allows us to focus on the logic of the code we want to test, it also has some drawbacks. In order to understand the state of this Git repository, we need to clone it and perform operations using `git` commands. Additionally, if we want to make small modifications, we need to use git commands, and it's not possible to review the changes made on GitHub since those changes are binary.

CleanShot 2024-05-30 at 12 57 55

To address these issues, this PR proposes using [go-git](https://github.com/go-git/go-git/) to dynamically create the test Git repository during the testing process. Although there is a potential downside that if the process of creating this test repository is incorrect, the tests will fail to run properly, the benefits outweigh this risk, IMO. Using Go to create the repository makes it clearer what branches are being created, what tags are being assigned, and so on. This approach also makes it easier to make changes and conduct reviews on GitHub.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/plugin/manager_unix_test.go [hunk 1]
```diff
@@ -14,36 +14,57 @@ import (
...
+	b = bytes.ReplaceAll(b, []byte("0.2.0"), []byte(version))
+	err = os.WriteFile(manifestPath, b, 0644)
 	require.NoError(t, err)
+}
+
+func TestManager_Install(t *testing.T) {
+	gs := setupGitRepository(t, "test_plugin", "testdata/test_plugin")
 	t.Cleanup(gs.Close)
 
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (DmitriyLewen) [inline pkg/plugin/manager_unix_test.go:70]: I detected one problem with this test server:
this zip file doesn't contain nested dir:
- test_plugin.zip/plugin.yaml

But GitHub archive contains `repo name dir`:
check https://github.com/aquasecurity/trivy-plugin-kubectl/archive/refs/heads/main.zip
- trivy-plugin-kubectl-main.zip/trivy-plugin-kubectl-main/plugin.yaml

I think we should handle this case, but perhaps in another PR.

File: pkg/plugin/manager_unix_test.go [hunk 2]
```diff
@@ -14,36 +14,57 @@ import (
...
+	r := gittest.Clone(t, gs, repo, worktree)
+
+	// git tag
+	gittest.SetTag(t, r, "v0.2.0")
+
+	// git commit
+	modifyManifest(t, worktree, "0.3.0")
+	gittest.CommitAll(t, r, "bump up to 0.3.0")
+
+	t.Log("git push")
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/plugin/manager_unix_test.go]: Deleted
https://github.com/aquasecurity/trivy/pull/6824/commits/25ad84c20c3570a966b8b30a2ab2fb1656a2694c

File: internal/gittest/server.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+var signature = &object.Signature{
+	Name:  "Test",
+	Email: "test@example.com",
+	When:  time.Now(),
+}
+
+func NewServer(t *testing.T, repo, dir string) *httptest.Server {
+	wtDir := t.TempDir()
+	fmt.Println(wtDir)
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply internal/gittest/server.go]: Deleted
https://github.com/aquasecurity/trivy/pull/6824/commits/dd56f1adf9fbdcd477fc6d71ba359f66ac074d13

File: pkg/plugin/manager_unix_test.go (hunk 1 above)
Comment (knqyf263) [reply pkg/plugin/manager_unix_test.go:69]: Good point. It's not introduced in this PR. I'll open another PR.

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Pull request merged**